src/modules/containers.nix

```
@@ -135,17 +135,19 @@ let
       }
     ];
 
-    config = {
-      Entrypoint = cfg.entrypoint;
-      User = "${user}";
-      WorkingDir = "${homeDir}";
-      Env = lib.mapAttrsToList
-        (name: value:
-          "${name}=${toString value}"
-        )
-        config.env ++ [ "HOME=${homeDir}" "USER=${user}" ];
-      Cmd = [ cfg.startupCommand ];
-    };
+    config = lib.attrsets.mergeAttrsList [
+      {
+        User = "${user}";
+        WorkingDir = "${homeDir}";
+      }
+      (if cfg.isDev then {
+        Env = lib.mapAttrsToList (name: value: "${name}=${toString value}")
+          config.env ++ [ "HOME=${homeDir}" "USER=${user}" ];
+        Entrypoint = cfg.entrypoint;
+        Cmd = [ cfg.startupCommand ];
+      } else
+        { })
+    ];
   };
```

Nitpick: A bit easier to read because it uses less control flow and we don't have conflicting keys here.

```suggestion
    config = {
      User = "${user}";
      WorkingDir = "${homeDir}";
    } // lib.optionalAttrs cfg.isDev {
      Env = lib.mapAttrsToList (name: value: "${name}=${toString value}")
          config.env ++ [ "HOME=${homeDir}" "USER=${user}" ];
      Entrypoint = cfg.entrypoint;
      Cmd = [ cfg.startupCommand ];
   };
```